app/src/main/java/org/wikipedia/diff/ArticleEditDetailsFragment.kt

```diff
@@ -322,46 +345,52 @@ class ArticleEditDetailsFragment : Fragment(), WatchlistExpiryDialog.Callback, L
         dialog.show()
     }
 
-    private fun createSpannable(diffs: List<DiffResponse.DiffItem>): CharSequence {
-        val spannableString = SpannableStringBuilder()
-        diffSize = 0
-        for (diff in diffs) {
-            val prefixLength = spannableString.length
-            spannableString.append(diff.text.ifEmpty { "\n" })
-            when (diff.type) {
-                DiffResponse.DIFF_TYPE_LINE_ADDED -> {
-                    diffSize += diff.text.length + 1
-                    updateDiffTextDecor(spannableString, true, prefixLength, prefixLength + diff.text.length)
-                }
-                DiffResponse.DIFF_TYPE_LINE_REMOVED -> {
-                    diffSize -= diff.text.length + 1
-                    updateDiffTextDecor(spannableString, false, prefixLength, prefixLength + diff.text.length)
-                }
-                DiffResponse.DIFF_TYPE_PARAGRAPH_MOVED_FROM -> {
-                    diffSize -= diff.text.length + 1
-                    updateDiffTextDecor(spannableString, false, prefixLength, prefixLength + diff.text.length)
-                }
-                DiffResponse.DIFF_TYPE_PARAGRAPH_MOVED_TO -> {
-                    diffSize += diff.text.length + 1
-                    updateDiffTextDecor(spannableString, true, prefixLength, prefixLength + diff.text.length)
+    private fun showUndoDialog() {
+        val dialog = UndoEditDialog(requireActivity())
+        dialog.callback = object : UndoEditDialog.Callback {
```

Comment:
We can add `fun` to the interface if it only has one method so that we can use lambda here.

```
    fun interface Callback {
        fun onSuccess(text: CharSequence)
    }
```

Reply:
yep!